### ISSUE TITLE
display-plugin-updates: Inherited plugin version not detected

### ISSUE THREAD

**Issue by kwin** [mojohaus/versions-maven-plugin#550]
When cloning https://github.com/apache/sling-parent and executing ` mvn org.codehaus.mojo:versions-maven-plugin:2.9.0:display-plugin-updates` it emits

```
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-jar-plugin ......................................... 3.2.2
```

for the module `sling-bundle-parent`, although the version of this plugin is managed in https://github.com/apache/sling-parent/blob/b7104da953b090821549d23f9bdf1ea1d002a1d8/sling-parent/pom.xml#L249 (the parent).

**Comment by stefanseifert** [comment]
it also happens with version 2.8.1 of the plugin.

strange issue. it works for other plugins which are declared in the same way.

from time to time i've observed this in very few other projects as well, e.g. https://github.com/wcm-io/io.wcm.maven.parent
in that context it seems to be related when the same plugin is also used in profiles without version number (which is correct but seems to be misinterpreted by the plugin). but it's not the same case here.

**Comment by kwin** [comment]
I have only seen this in projects where the pom.xml in the parent folder is not the parent pom.xml, i.e. separated reactor and parent, but in the example from above the reference to the parent is definitely correct (https://github.com/apache/sling-parent/blob/b7104da953b090821549d23f9bdf1ea1d002a1d8/sling-bundle-parent/pom.xml#L22-L25).

**Comment by ajarmoniuk** [comment]
I have been looking into this, trying to get to the root cause of the issue. I managed to reproduce the problem with https://github.com/apache/sling-parent/tree/sling-parent-reactor-47 and plugin version 2.8.1 through 2.9.0 after which #550 effectively hid the issue.
 
However, the problem seems to have been helped with with the latest snapshot (if I remove Stefan's patches):

```
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   maven-jar-plugin ................................... 3.2.2 -> 3.3.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
```

It appears that it's already been fixed at 2.10.0 with the patch reverted!

```
[INFO] --- versions-maven-plugin:2.10.0:display-plugin-updates (default-cli) @ sling-bundle-parent ---
[INFO] 
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   maven-jar-plugin ................................... 3.2.2 -> 3.3.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
[INFO] 
[INFO] All plugins have a version specified.
[INFO] 
[INFO] Project requires minimum Maven version for build of: 3.3.9
[INFO] Plugins require minimum Maven version of: 3.3.9
```

But it was of course there with 2.9.0:

```
[INFO] --- versions-maven-plugin:2.9.0:display-plugin-updates (default-cli) @ sling-bundle-parent ---
[INFO] 
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
[INFO] 
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-jar-plugin ......................................... 3.3.0
[INFO] 
[INFO] Project requires minimum Maven version for build of: 3.3.9
[INFO] Plugins require minimum Maven version of: 3.3.9
[INFO] 
[INFO] No plugins require a newer version of Maven than specified by the pom.
```

It's highly possible that the issue was already have been resolved when #550 was merged.
